### ISSUE TITLE
[BUG] Duplicate Test Results Displayed

### ISSUE THREAD

**Issue by clint156** [rokwire/safer-illinois-app#94]
I got tested on 8/25 and 8/28. I received one notification for each test yet both show up twice in my history. Melvin has the same issue.

**Comment by mihail-varbanov** [comment]
@clint156 , how many devices do you have with Safer app installed and being logged in with your account?

**Comment by clint156** [comment]
I have two devices with it installed. One is an ipad that I rarely use for some testing. When I got my results yesterday, the result only showed up once.


Clint Stearns
Systems Development Architect, Student Affairs Technology

[Illinois-Wordmark-Horizontal-Full-Color-RGB_185px]

**Comment by pmarkhennessy** [comment]
@clint156 is this still an issue and please answer Misho's question on devices used please

**Comment by clint156** [comment]
@pmarkhennessy I thought I did answer the question on devices used, at least it shows a response in the comments above, or did I reply to the email incorrectly? Anyway, the answer is two, one ios and one android, although the IOS device hasn't been on in a while. It was an issue with two tests (8/28 and 8/25), seemed to go away with my 9/1 and 9/4 tests, but duplicated my 9/7 tests again. That would have been with the build prior to 2.5.7.

**Comment by mihail-varbanov** [comment]
@pmarkhennessy, yes, Clint did answer my question by email. I succeed to recreate the issue as well. I believe it is caused when there are multiple devices having the Safer app installed logged in to the same account due to the concurrency when processing pending events/test results. In short, we do not have a useful move in this situation.

**Comment by mihail-varbanov** [comment]
@pmarkhennessy, yes, Clint did answer my question by email. I succeed to recreate the issue as well. I believe it is caused when there are multiple devices having the Safer app installed logged in to the same account due to the concurrency when processing pending events/test results. In short, we do not have a useful move in this situation.

**Comment by mihail-varbanov** [comment]
@pmarkhennessy, yes, Clint did answer my question by email. I succeed to recreate the issue as well. I believe it is caused when there are multiple devices having the Safer app installed logged in to the same account due to the concurrency when processing pending events/test results. In short, we do not have a useful move in this situation.

**Comment by mihail-varbanov** [comment]
@pmarkhennessy, yes, Clint did answer my question by email. I succeed to recreate the issue as well. I believe it is caused when there are multiple devices having the Safer app installed logged in to the same account due to the concurrency when processing pending events/test results. In short, we do not have a useful move in this situation.

**Comment by mihail-varbanov** [comment]
@pmarkhennessy, yes, Clint did answer my question by email. I succeed to recreate the issue as well. I believe it is caused when there are multiple devices having the Safer app installed logged in to the same account due to the concurrency when processing pending events/test results. In short, we do not have a useful move in this situation.

**Comment by mihail-varbanov** [comment]
@pmarkhennessy, yes, Clint did answer my question by email. I succeed to recreate the issue as well. I believe it is caused when there are multiple devices having the Safer app installed logged in to the same account due to the concurrency when processing pending events/test results. In short, we do not have a useful move in this situation.

**Comment by mihail-varbanov** [comment]
I found a gap in client logic that could allow such concurrent processing within a single client application. This must be fixed in 2.7.3 and later versions.

It is quite complex to handle the case when 2 devices process pending events in parallel (e.g. after receiving FCM). I have some ideas about this and will try to resolve it in 2.8.
